### DISCUSSION TITLE
Support for OpenAPI authentication documentation

### DISCUSSION THREAD

**Discussion by ctrl-Felix** [litestar-org/litestar#1813] [Q&A] [answered]
The OpenAPI schema supports definining authentication and protectecd API endpoints:
https://swagger.io/docs/specification/authentication/

Now I was wondering if Litestar already supports that, because I was not able to find/activate it. So if it's supported I'd be very grateful if someone could point me out how to do it. 

Otherwise I think it would make sense to add that.

**Comment by JacobCoffee** [comment]
I believe this was answered in Discord - https://discord.com/channels/919193495116337154/919193495690936353/1118449174749388880

**Reply by ctrl-Felix** [reply]
The answer was referring to the JWT contrib package. I am referring to the AbstractAuthenticationMiddleware

**Reply by Goldziher** [reply, marked as answer]
the `AbstractAuthenticationMiddleware` is an abstract class that needs to be implemented in a concrete way. Once you implement an authentication backend, you can configure your openapi specs to include authentication documentation for this. 

An example of how to do this can be found here: https://github.com/litestar-org/litestar/blob/main/litestar/contrib/jwt/jwt_auth.py#L68

There is no way for us to automagically know what you implemented.
